File: datafile-handler/src/test/java/com/optimizely/ab/android/datafile_handler/DefaultDatafileHandlerUnitTest.java
package com.optimizely.ab.android.datafile_handler;

import android.content.Context;

import com.optimizely.ab.android.shared.DatafileConfig;

import org.junit.Test;

import static org.junit.Assert.assertFalse;
import static org.mockito.Mockito.mock;
import static org.mockito.Mockito.when;

/**
 * Example local unit test, which will execute on the development machine (host).
 *
 * @see <a href="http://d.android.com/tools/testing">Testing documentation</a>
 */
public class DefaultDatafileHandlerUnitTest {

    DatafileHandler handler = mock(DefaultDatafileHandler.class);

    @Test
    public void testHandler() throws Exception {
        handler = new DefaultDatafileHandler();
        Context context = mock(Context.class);
        when(context.getApplicationContext()).thenReturn(context);
        assertFalse(handler.isDatafileSaved(context, new DatafileConfig("1", null)));
    }
}

File: datafile-handler/src/test/java/com/optimizely/ab/android/datafile_handler/DefaultDatafileHandlerUnitTest.kt
package com.optimizely.ab.android.datafile_handler

import android.content.Context
import com.optimizely.ab.android.datafile_handler.DefaultDatafileHandler
import com.optimizely.ab.android.shared.DatafileConfig
import org.junit.Assert
import org.junit.Test
import org.junit.runner.RunWith
import org.mockito.Mockito
import org.powermock.api.mockito.PowerMockito
import org.powermock.core.classloader.annotations.PrepareForTest
import org.powermock.modules.junit4.PowerMockRunner

/**
 * Example local unit test, which will execute on the development machine (host).
 *
 * @see [Testing documentation](http://d.android.com/tools/testing)
 */
@RunWith(PowerMockRunner::class)
@PrepareForTest(DefaultDatafileHandler::class)
class DefaultDatafileHandlerUnitTest {
    var handler: DatafileHandler = PowerMockito.mock(DefaultDatafileHandler::class.java)
    @Test
    @Throws(Exception::class)
    fun testHandler() {
        handler = DefaultDatafileHandler()
        val context = PowerMockito.mock(Context::class.java)
        PowerMockito.`when`(context.applicationContext).thenReturn(context)
        Assert.assertFalse(handler.isDatafileSaved(context, DatafileConfig("1", null))!!)
    }
}